quaddtype/numpy_quaddtype/src/ops.hpp

```diff
@@ -0,0 +1,513 @@
+{
+    *out = asinl(*op);
+    return 0;
+}
+
+static inline int
+ld_acos(long double *op, long double *out)
+{
+    *out = acosl(*op);
+    return 0;
+}
+
+static inline int
+ld_atan(long double *op, long double *out)
+{
+    *out = atanl(*op);
+    return 0;
+}
+
+// Binary Quad operations
+typedef int (*binary_op_quad_def)(Sleef_quad *, Sleef_quad *, Sleef_quad *);
+
+static inline int
+quad_add(Sleef_quad *out, Sleef_quad *in1, Sleef_quad *in2)
+{
+    *out = Sleef_addq1_u05(*in1, *in2);
+    return 0;
+}
+
+static inline int
+quad_sub(Sleef_quad *out, Sleef_quad *in1, Sleef_quad *in2)
+{
+    *out = Sleef_subq1_u05(*in1, *in2);
+    return 0;
+}
+
+static inline int
+quad_mul(Sleef_quad *res, Sleef_quad *a, Sleef_quad *b)
+{
+    *res = Sleef_mulq1_u05(*a, *b);
+    return 0;
+}
+
+static inline int
+quad_div(Sleef_quad *res, Sleef_quad *a, Sleef_quad *b)
+{
+    *res = Sleef_divq1_u05(*a, *b);
+    return 0;
+}
+
+static inline int
+quad_pow(Sleef_quad *res, Sleef_quad *a, Sleef_quad *b)
+{
+    *res = Sleef_powq1_u10(*a, *b);
+    return 0;
+}
+
+static inline int
+quad_mod(Sleef_quad *res, Sleef_quad *a, Sleef_quad *b)
+{
+    *res = Sleef_fmodq1(*a, *b);
+    return 0;
+}
+
+static inline int
+quad_minimum(Sleef_quad *out, Sleef_quad *in1, Sleef_quad *in2)
+{
+    *out = Sleef_icmpleq1(*in1, *in2) ? *in1 : *in2;
+    return 0;
+}
+
+static inline int
+quad_maximum(Sleef_quad *out, Sleef_quad *in1, Sleef_quad *in2)
+{
+    *out = Sleef_icmpgeq1(*in1, *in2) ? *in1 : *in2;
+    return 0;
+}
+
+static inline int
+quad_atan2(Sleef_quad *out, Sleef_quad *in1, Sleef_quad *in2)
+{
+    *out = Sleef_atan2q1_u10(*in1, *in2);
+    return 0;
+}
+
+// Binary long double operations
+typedef int (*binary_op_longdouble_def)(long double *, long double *, long double *);
+
+static inline int
+ld_add(long double *out, long double *in1, long double *in2)
+{
+    *out = (*in1) + (*in2);
+    return 0;
+}
+
+static inline int
+ld_sub(long double *out, long double *in1, long double *in2)
+{
+    *out = (*in1) - (*in2);
+    return 0;
+}
+
+static inline int
+ld_mul(long double *res, long double *a, long double *b)
+{
+    *res = (*a) * (*b);
+    return 0;
+}
+
+static inline int
+ld_div(long double *res, long double *a, long double *b)
+{
+    *res = (*a) / (*b);
+    return 0;
+}
+
+static inline int
+ld_pow(long double *res, long double *a, long double *b)
+{
+    *res = powl(*a, *b);
+    return 0;
+}
+
+static inline int
+ld_mod(long double *res, long double *a, long double *b)
+{
+    *res = fmodl(*a, *b);
+    return 0;
+}
+
+static inline int
+ld_minimum(long double *out, long double *in1, long double *in2)
+{
+    *out = (*in1 < *in2) ? *in1 : *in2;
+    return 0;
+}
+
+static inline int
+ld_maximum(long double *out, long double *in1, long double *in2)
+{
+    *out = (*in1 > *in2) ? *in1 : *in2;
+    return 0;
+}
+
+static inline int
+ld_atan2(long double *out, long double *in1, long double *in2)
+{
+    *out = atan2l(*in1, *in2);
+    return 0;
```

Comment:
Is it possible for any of the functions from here up implementing various operations to ever return anything but 0? I don't see anything... In that case it probably makes more sense to just return the result of the operation and not have unncessary error codes.

Reply:
This is resolved